modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientCache.java

```diff
@@ -947,4 +1061,73 @@ private void writeEntries(Map<? extends K, ? extends V> map, PayloadOutputChanne
                     serDes.writeObject(out, e.getValue());
                 });
     }
+
+    /**
+     * Adapter to convert CQ listener calls to JCache listener calls.
+     */
+    private static class JCacheEntryListenerAdapter<K, V> implements CacheEntryUpdatedListener<K, V> {
```

Comment:
The file is too big already, can we move those classes to separate files?

Reply:
I've moved `JCacheEntryListenerAdapter` to the separate file, but `JCacheDisconnectListenerAdapter` is not static and relies on `TcpClientCache` fields, so I've inlined it as an anonymous class.